common/lib/container-definitions/src/loader.ts

```diff
@@ -148,6 +148,16 @@ export namespace ConnectionState {
  */
 export type ConnectionState = ConnectionState.Disconnected | ConnectionState.Connecting | ConnectionState.Connected;
 
+/**
+ * Accepted reconnection modes to be used with `setAutoReconnect()`
+ */
+export enum ReconnectMode {
```

Comment:
why are we moving this to common? that seems really odd.

Reply:
i don't think these changes should be coupled

Reply:
The motivation behind moving it is based on https://github.com/microsoft/FluidFramework/issues/9167. To summarize, we want to rework `setAutoReconnect()` and have it use `ReconnectMode` as the first argument in the function. We also want to expose `setAutoReconnect()` so the developer will need to have access to `ReconnectMode`.

Reply:
i wouldn't use an enum here. we should just use `export type ReconnectMode= ...` and let the consumer bring their own strings

Reply:
would be great it the spec were in the design item: https://github.com/microsoft/FluidFramework/issues/8745

Reply:
if you just add the type. you could keep the enum at the loader layer, probably don't need to export it

Reply:
Are there any other advantages of using a type vs an enum in this situation (other than not having to move the current `ReconnectMode` enum)?

Reply:
enums emit javascript code. types do not. i'd like our definition packages to eventually be types only